src/compiler/checker.ts

```diff
@@ -4798,7 +4798,8 @@ namespace ts {
                     }
                     function shouldWriteTypeOfFunctionSymbol() {
                         const isStaticMethodSymbol = !!(symbol.flags & SymbolFlags.Method) &&  // typeof static method
-                            some(symbol.declarations, declaration => hasSyntacticModifier(declaration, ModifierFlags.Static));
+                            some(symbol.declarations, declaration =>
+                                hasSyntacticModifier(declaration, ModifierFlags.Static) && isMethodDeclaration(declaration) && !isComputedPropertyName(declaration.name));

```

Comment:
what happens with property declarations? does that have an existing test?

Reply:
Moreover, when the static member with a computed name that we're referencing has a `unique symbol` type itself, what do we do? I expect we'd need to emit `(typeof Foo)[key]`.

Reply:
@sandersn I've added additional tests. 

@weswigham `(typeof Foo)[key]` is this allowed in declarations? Wouldn't that throw an error like
`"'key' refers to a value, but is being used as a type ...."`?

Reply:
Well... `(typeof Foo)[typeof key]` :3